index.d.ts

```diff
@@ -0,0 +1,45 @@
+/// <reference types="node" />
+
+import { FastifyPlugin, FastifyRequest } from 'fastify';
+
+declare module 'fastify' {
+  interface FastifyInstance {
+    csrfProtection(req: any, reply: any, done: any): any;
```

Comment:
@Ethan-Arrowood @fox1t this is a generic hook handler, I'm not entirely sure how to type it.
Any suggestion?

Reply:
You should probably give the req/res arguments the corresponding FastifyRequest and FastifyReply types. Other than that this is looking good

Reply:
Thanks!